### PR TITLE
New Post: Cimpler -- continous integration, unix style

### PR DESCRIPTION
**Pull request by danielbeardsley**
A post introducing the CI server that iFixit has been using for 6 months.

### CODE REVIEW

File: _posts/2013-03-03-cimpler-continuous-integration-server.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+Because the world really needed another CI server, we made one.
+Checkout [cimpler on github][repo] and feel free to contribute.
+
+At [iFixit](http://www.ifixit.com), we spent far too long
+trying to configure [Jenkins](http://jenkins-ci.org/)
+and its many plugins _just right_.
+Jenkins and friends ended up being unintuitive (at least for our purposes)
+and had far more features than we needed.
+
+I love node.js and needed something simple, single-purpose, and extensible,
```
Comment (timothyasp) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md:19]: What about a dash at the end to add emphasis to "you know, unix-style". http://lilt.ilstu.edu/golson/punctuation/dash.html

"I love node.js and needed something simple, single-purpose, and extensivle - you know, unix-style."

**Comment by timothyasp** [comment]
Reads well and does a good job explaining cimpler - great work!

File: _posts/2013-03-03-cimpler-continuous-integration-server.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+---
+layout: with-comments
+title: "Cimpler - continuous integration, unix style"
+author: Daniel Beardsley
+author_url: http://github.com/danielbeardsley
+summary: Cimpler is a continuous intergration server in node.js
```
Comment (marczych) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: `integration`

"integration server **written** in node.js" ?

File: _posts/2013-03-03-cimpler-continuous-integration-server.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+         that aims to do one this, and do it well, or at least simply.
```
Comment (marczych) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: "to do one **thing**"?

File: _posts/2013-03-03-cimpler-continuous-integration-server.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+and it's been rock solid
+(we haven't even bothered to move it out of a screen session).
+
+## How iFixit Uses Cimpler
+
+1. The github plugin adds builds to the queue
+when it receives a [post-receive][hooks]
+notification from github.
+2. The git-build plugin takes builds off the queue,
+does some merging,
```
Comment (marczych) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Should we explain what merging is done and why?

File: _posts/2013-03-03-cimpler-continuous-integration-server.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+$ # symlink the commandline interface
+$ ln -s `pwd`/bin/cimpler /usr/local/bin/
+$ node server.js >/var/log/cimpler.log &
+$ cimpler --help
+{% endhighlight %}
+
+## What it does
+
+**Cimpler is mostly just a queue with some plugins**
+with a few useful special-purpose constructs provided for its plugins to utilize.
```
Comment (marczych) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Two `with`s in the same sentence sounds kinda awkward.

File: _posts/2013-03-03-cimpler-continuous-integration-server.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+option.
+
+Configuration is done via a javascript file that exports a JS object.
+This is the entire configuration file we use at iFixit.
+
+{% highlight javascript %}
+module.exports = {
+   /**
+    * All plugin can register connect middleware that
+    * will listen on this port.
```
Comment (marczych) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md:96]: `plugins`

Do we need both `register` and `connect`?

File: _posts/2013-03-03-cimpler-continuous-integration-server.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+layout: with-comments
+title: "Cimpler - continuous integration, unix style"
+author: Daniel Beardsley
+author_url: http://github.com/danielbeardsley
+summary: Cimpler is a continuous intergration server in node.js
+         that aims to do one this, and do it well, or at least simply.
+---
+
+Because the world really needed another CI server, we made one.
+Checkout [cimpler on github][repo] and feel free to contribute.
```
Comment (xiongchiamiov) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: [Checkout](http://www.merriam-webster.com/dictionary/checkout) vs. [check out](http://www.merriam-webster.com/dictionary/check+out) - both are valid nouns _and_ verbs, but with slightly different meanings.  Stupid English.

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 5 above)
Comment (xiongchiamiov) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Could just change the second one to an "and": Cimpler is mostly just a queue with some plugins and a few useful constructs...

File: _posts/2013-03-03-cimpler-continuous-integration-server.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+         logs: {
+            path: '/var/www/pub/cimpler-logs/',
+            url:  'http://[REDACTED]/cimpler-logs/'
+         },
+         timeout: 20 * 60 * 1000 // 20 minutes
+      },
+
+      'github-commit-status': {
+         auth: {
+            // Anything accepted github.authenticate()
```
Comment (xiongchiamiov) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Anything accepted _in_?

File: stylesheets/styles.css [hunk 9]
```diff
@@ -9,11 +9,21 @@ body {
...
-  margin:0 0 20px;
+  margin:0 0 15px;
 }
 
 p, ul, ol, table, pre, dl {
-  margin:0 0 20px;
+  margin:0 0 25px;
+}
+
+/* Make litsts a little closer to their preceeding paragraphs */
```
Comment (xiongchiamiov) [inline stylesheets/styles.css]: [*lists](http://harkavagrant.com/index.php?id=302)

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 6 above)
Comment (danielbeardsley) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md:96]: Thanks, but connect is being used as a [noun](http://www.senchalabs.org/connect), not a verb, so `connect.js` it is. `plugin` is fixed though.

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 1 above)
Comment (danielbeardsley) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md:19]: Fixed.

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 2 above)
Comment (danielbeardsley) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Fixed... slightly different.

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 3 above)
Comment (danielbeardsley) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Fixed

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 4 above)
Comment (danielbeardsley) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Fixed.

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 5 above)
Comment (danielbeardsley) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Fixed

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 7 above)
Comment (danielbeardsley) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Thanks! Fixed.

File: _posts/2013-03-03-cimpler-continuous-integration-server.md (hunk 8 above)
Comment (danielbeardsley) [inline _posts/2013-03-03-cimpler-continuous-integration-server.md]: Fixed.

File: stylesheets/styles.css (hunk 9 above)
Comment (danielbeardsley) [inline stylesheets/styles.css]: Fixed.

**Pull request merged**